Sources/MobileCoinClient.swift

```diff
@@ -212,11 +212,16 @@ public final class MobileCoinClient {
         memoType: MemoType = .recoverable,
         amount: Amount,
         fee: UInt64,
-        rngSeed: RngSeed,
+        rng: MobileCoinRng,
         completion: @escaping (
             Result<PendingSinglePayloadTransaction, TransactionPreparationError>
         ) -> Void
     ) {
+        guard let rngSeed = rng.generateRngSeed() else {
```

Comment:
not sure this is correct, if you only use the RNG to create a seed which you then pass around to see our internal RNG, that defeats the purpose of letting them pass in their own RNG class

Reply:
Android uses the input Rng to seed the RNG for output idempotence.   So here, we are doing the same as Android with respect to the output transaction creation.

Reply:
well then, doesnt seem to be as flexible as I initially thought, but if its the same as android thats correct